userspace/libsinsp/sinsp_filtercheck_container.cpp

```diff
@@ -499,6 +517,34 @@ uint8_t *sinsp_filter_check_container::extract_single(sinsp_evt *evt,
 			m_val.u32 = (container_info->m_privileged ? 1 : 0);
 		}
 
+		RETURN_EXTRACT_VAR(m_val.u32);
+		break;
+	case TYPE_CONTAINER_HOST_PID:
+	case TYPE_CONTAINER_HOST_NETWORK:
+	case TYPE_CONTAINER_HOST_IPC:
+		if(tinfo->m_container_id.empty()) {
```

Comment:
Can't we use `is_host` here?

Reply:
Done! As a side note, I initially skipped using this because it uses the pid_namespace
```
bool is_host = tinfo->m_container_id.empty() && !tinfo->is_in_pid_namespace();
```
That part is not 100% true in my opinion, but I think it's there for a reason (skipped execve events where we miss container_id coming from the group?)

Reply:
This is when that check was modified last time https://github.com/falcosecurity/libs/pull/1604
@jasondellaluce I agree with your fix but at the same time, I'm not feeling 100% convinced since containers can be started sharing namespaces with the host.
Namespaces are shared, but cgroups should always be there, and our extraction of the container id too IMO.
Otherwise, something that will work in 99% of the cases would check also for the other shareable namespace here (will not work when all the three of them are shared, which is rare but possible)

Reply:
/hold for @jasondellaluce response

Reply:
> /hold for @jasondellaluce response

👍 

anyway, I believe any further fix regarding `is_host` should be addressed by a new PR.